backend/service/martini/orgs.go

```diff
@@ -32,3 +38,171 @@ func (s *Service) GetOrgsHandler() rest.Handle {
 		marshal.WriteJSON(w, r, res)
 	}
 }
+
+// ValidateOrgHandler validates Org registration
+func (s *Service) ValidateOrgHandler() rest.Handle {
+	return func(w http.ResponseWriter, r *http.Request, p rest.Params) {
+		req := new(v1.ValidateOrgRequest)
+		err := marshal.DecodeBody(w, r, req)
+		if err != nil {
+			return
+		}
+
+		t, err := s.db.UseApprovalToken(r.Context(), req.Token, req.Code)
+		if err != nil {
+			marshal.WriteJSON(w, r, httperror.WithInvalidRequest("the code is not valid or already validated").WithCause(err))
+			return
+		}
+
+		org, err := s.db.GetOrg(r.Context(), t.OrgID)
+		if err != nil {
+			marshal.WriteJSON(w, r, httperror.WithUnexpected("unable to find organization").WithCause(err))
+			return
+		}
+
+		org.Status = string(v2acme.StatusValid)
+		org, err = s.db.UpdateOrgStatus(r.Context(), org)
```

Comment:
@dissoupov so when multiple requestor request same Org, the very first one will just update org status? Can there be some racing issues?

Reply:
that flow we need to finish. for now only the first requestor locks the org